x-pack/plugins/ml/public/ui_actions/open_in_anomaly_explorer_action.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ActionContextMapping, createAction } from '../../../../../src/plugins/ui_actions/public';
+import {
+  AnomalySwimlaneEmbeddable,
+  SwimLaneDrilldownContext,
+} from '../embeddables/anomaly_swimlane/anomaly_swimlane_embeddable';
+import { MlCoreSetup } from '../plugin';
+import { ML_APP_URL_GENERATOR } from '../url_generator';
+
+export const OPEN_IN_ANOMALY_EXPLORER_ACTION = 'openInAnomalyExplorerAction';
+
+export function createOpenInExplorerAction(getStartServices: MlCoreSetup['getStartServices']) {
+  return createAction<typeof OPEN_IN_ANOMALY_EXPLORER_ACTION>({
+    id: 'open-in-anomaly-explorer',
+    type: OPEN_IN_ANOMALY_EXPLORER_ACTION,
+    getIconType(context: ActionContextMapping[typeof OPEN_IN_ANOMALY_EXPLORER_ACTION]): string {
+      return 'tableOfContents';
+    },
+    getDisplayName() {
+      return i18n.translate('xpack.ml.actions.openInAnomalyExplorerTitle', {
+        defaultMessage: 'Open in Anomaly Explorer',
+      });
+    },
+    async getHref({ embeddable, data }: SwimLaneDrilldownContext): Promise<string> {
```

Comment:
The page selection isn't being retained on drilldown here:

![swimlane_drilldown](https://user-images.githubusercontent.com/7405507/87298811-bbd31200-c502-11ea-831f-3011fd116b38.gif)

Reply:
Couldn't reproduce this issue. Could you please verify if it's still the case after pulling the latest changes?

Reply:
I can still reproduce this with the latest changes. Works first time I try, but then always opens up to show the first page,

Reply:
There was flakiness because of async job selection from the app state clashing with the restored state. Fixed in https://github.com/elastic/kibana/pull/71082/commits/5fa4baf276770f907eacffbaf42ce061b3f279be. But we might want to revisit the logic of the `appState` and `globalState` later on.

Reply:
Page selection is working for me on drilldown with the latest edits.